program_intent_engagement/apps/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,139 @@
+"""
+Tests for the program intent API views
+"""
+import json
+import uuid
+
+import ddt
+from django.urls import reverse
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+from test_utils import ProgramIntentsAPITestCase
+
+
+@ddt.ddt
+class ProgramIntentsViewTests(ProgramIntentsAPITestCase):
+    """
+    Tests Program Intents View
+    """
+
+    def setUp(self):
+        super().setUp()
+
+        self.url = reverse('api:v1:program_intent-insert')
+
+        self.program_uuid = str(uuid.uuid4())
+        self.data = {
+            'program_uuid': self.program_uuid,
+            'reason': 'ALL_COURSES_COMPLETE',
+            'certainty': 'CERTAIN_YES',
+            'effective_timestamp': '2022-07-01 00:00:00',
+        }
+
+    def post_api(self, user, data):
+        """
+        Helper function to make a POST request to the API
+        """
+        data = json.dumps(data)
+        headers = self.build_jwt_headers(user)
+
+        return self.client.post(self.url, data, **headers, content_type="application/json")
+
+    def get_response(self, user, data, expected_response):
```

Comment:
naming nits - expected_response should be called expected response code or expected code, expected response is the name for a full response object

this function should be called something like validate response code because it is not just retrieving data